File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> c;
        int l = 0;
        int r = 0;
        int start = 0;
        while(nums[l] != nums[r] || start == 0) {
            start = 1;
            if (l < n - 1){
                l += 1;
            }
            else {
                l = 0;
            }
            if (r < n - 2) {
                r += 2;
            }
            else if (r < n -1) {
                r = 0;
            }
            else {
                r = 1;
            }

        }   
        return nums[l];     
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        l = 0
        r = 0
        for i in range(len(nums)):
            r = i
            
            while r < len(nums) and abs(r - i) <= k:
                if nums[r] == nums[i] and r != i:
                    return True
                else:
                    r += 1
        return False     
